Backend/Middleware/src/main/java/middleware/repository/JWTokenRepository.java

```diff
@@ -0,0 +1,32 @@
+package middleware.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.query.Procedure;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import middleware.model.JWToken;
+
+/**
+ * Repository interface for performing CRUD operations on {@link JWToken} entities.
+ * This interface extends {@link JpaRepository}, providing built-in methods for
+ * managing {@link JWToken} objects in the database.
+ *
+ * Additionally, this repository includes methods for executing stored procedures
+ * related to JSON Web Token (JWT) management, such as updating the last login,
+ * logging out, and checking JWT validity.
+ *
+ * The methods in this repository use the {@link Procedure} annotation to specify
+ * the corresponding stored procedures in the database.
+ */
+@Repository
+public interface JWTokenRepository extends JpaRepository<JWToken, String> {
+
+        /**
+     * Checks the if JWT exists and is valid.
```

Comment:
think can add one more line as to where it checks for (i think this is database right)